src/python/pants/backend/python/tasks/pytest_run.py

```diff
@@ -514,8 +516,14 @@ def _do_run_tests_with_args(self, pex, args):
       if profile:
         env['PEX_PROFILE_FILENAME'] = '{0}.subprocess.{1:.6f}'.format(profile, time.time())
 
+      # Ensure the pex is invoked with the interpreter we selected beforehand to be compatible with
+      # all the relevant targets.
+      interpreter = self.context.products.get_data(PythonInterpreter)
+      env['PEX_PYTHON'] = interpreter.binary
```

Comment:
Te pexrc issue seems out of hand and this seems like a bandaid.

Is this more appropriate?
```
$ pex --help-variables
...
PEX_IGNORE_RCFILES: Boolean

    Explicitly disable the reading/parsing of pexrc files (~/.pexrc).
    Default: false.
...
```
It seems like we should be setting this when invoking any pex-based tool in pants, not just here.

Reply:
I thought that flag was on its way out: https://github.com/pantsbuild/pex/pull/673, but if it's okay to use it, then it looks like the thing we want to use.

Reply:
Yes, the flag, not the option. Flags for pex like --rcfile are only used at build time. options like PEX_IGNORE_RCFILES at runtime. For some options, specifying them at build time or run time both make sense. For this one, only run time makes sense.

Reply:
Thanks for the clarification!

Reply:
The flag `PEX_IGNORE_RCFILES` has not appeared to have an effect on the case I was looking at (and the case I was looking at didn't appear to be due to the use of a pexrc!). I was able to isolate a repro and convert it into the new `test_succeeds_for_intersecting_unique_constraints()`, which this PR now fixes.